hicommonwealth/commonwealth
# Integrate Axie Infinity.

## Description
Adds support for Axie Infinity, with a new SSO flow, including SsoToken table in the database which manages both magic and ronin logins.

## How to test
1. Make sure the server migrations have been run and succeeded
2. Get an Axie shared secret from me, for your .env, and install Ronin wallet and create an account, enabled for use on incognito mode.
3. Add the following line to /etc/hosts: `127.0.0.1  commonwealth.im`, to forward links to commonwealth.im to your localhost. **Remember to disable this once you are finished testing.**
4. Run `sudo local-ssl-proxy --source 443 --target 8080`, to forward ssl requests to the commonwealth localhost server.
5. Start the server as normal `yarn start` and visit, in an incognito tab, https://commonwealth.im/axie-infinity, which should ask you to confirm the unsafe cert. Do this and the page should load.
6. Try to log in using "Ronin Wallet" -- it will redirect you to Axie's page where you will sign a message, and then it should redirect you back and "just work".

## Have proper tags been added (for bug, enhancement, breaking change)?

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [x] yes, but I did not run tests
